rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/util/WorkflowVariablesTransformer.java

```diff
@@ -38,6 +38,8 @@
     public Map<String, String> getWorkflowVariablesFromPathSegment(PathSegment pathSegment) {
         Map<String, String> variables = null;
         MultivaluedMap<String, String> matrixParams = pathSegment.getMatrixParameters();
+        // Remove any empty keys that might be mistakenly sent to the scheduler to prevent bad behaviour
+        matrixParams.remove("");
```

Comment:
please move this statement inside the if, as matricParams can be null

Reply:
@fviale In fact this can never be null because the pathSegment.getMatrixParameters() returns always returns a map, even an empty one.
btw sonar insisted that I remove the !=null test but I didn't want to touch that

Reply:
sure, but even if it cannot be null, you need to put this remove inside the check, otherwise the code  is not readable, removing makes no sense on a null or empty map.